x-pack/plugins/endpoint/public/applications/endpoint/view/alerts/resolver.tsx

```diff
@@ -12,21 +12,26 @@ import { Resolver } from '../../../../embeddables/resolver/view';
 import { EndpointPluginServices } from '../../../../plugin';
 import { LegacyEndpointEvent } from '../../../../../common/types';
 import { storeFactory } from '../../../../embeddables/resolver/store';
+import { useAlertListSelector } from './hooks/use_alerts_selector';
+import * as selectors from '../../store/alerts/selectors';
 
 export const AlertDetailResolver = styled(
-  React.memo(
-    ({ className, selectedEvent }: { className?: string; selectedEvent?: LegacyEndpointEvent }) => {
-      const context = useKibana<EndpointPluginServices>();
-      const { store } = storeFactory(context);
-      return (
-        <div className={className} data-test-subj="alertResolver" data-testid="alertResolver">
-          <Provider store={store}>
-            <Resolver selectedEvent={selectedEvent} />
-          </Provider>
-        </div>
-      );
+  React.memo(({ className }: { className?: string }) => {
```

Comment:
consider taking in a LegacyEndpointEvent. The component that renders this could get `selectedAlertDetailsData` and then check if its a LegacyEndpointEvent via a predicate function and then pass it down. that way there is no unsafe casting